gr-utils/modtool/core/rm.py

```diff
@@ -105,13 +100,28 @@ def _remove_py_test_case(filename=None, ed=None):
             pbdoc_files_deleted = self._run_subdir('python/bindings/docstrings', ('*.h',), ('',))
 
             # Update python_bindings.cc
-            ed = CPPFileEditor(self._file['ccpybind'])
-            ed.remove_value('// BINDING_FUNCTION_PROTOTYPES(', '// ) END BINDING_FUNCTION_PROTOTYPES', 
-                'void bind_' + self.info['blockname'] + '(py::module& m);')
-            ed.remove_value('// BINDING_FUNCTION_CALLS(', '// ) END BINDING_FUNCTION_CALLS', 
-                'bind_' + self.info['blockname'] + '(m);')
-            ed.write()
+            blocknames_to_delete = []
+            if self.info['blockname']:
+                # A complete block name was given
+                blocknames_to_delete.append(self.info['blockname'])
+            elif self.info['pattern']:
+                # A regex resembling one or several blocks were given
+                blocknames_to_delete = get_block_names(self.info['pattern'], self.info['modname']) 
```

Comment:
Are we deleting all blocks matching a regex, this is too unsafe.

Reply:
this was the original feature of gr_modtool. Any argument given as stdin is taken as a regex and deletes all those blocks. I merely kept it in

Reply:
Im not sure if this is the actual intention of gr_modtool, but it seems to me that when a block name is given as an argument (e.g. `gr_modtool rm my_block`), it means the user wants that specific block to be acted on. On the other hand, when a regex is given as a stdin (afterall the prompt is for a regex), then the user wants a few blocks matching the regex to be acted on. So I didnt want to change the behavior of the modtool as it is.

However, I noticed that modtool does not behave exactly as its supposed to, so Ive made a fix in PR #4709. The issue and solution are described in greater detail there. Could I trouble you to review that too, please?

Reply:
Yeah, got your point, I think let's give the prompt for actual blockname, we can show the probable candidates if user enters incorrectly. Deletion based on regex can basically remove everything due to a single mistake, or do we notify before for confirmation, I forgot the behaviour?

Reply:
Yep. There is a confirmation before it actually deletes "Really delete include/mod_name/block_name.h? [Y/n/a/q]: "

Reply:
Cool, then